src/lib/util/RunOutputProcessor.ts

```diff
@@ -28,21 +28,11 @@ export class RunOutputProcessor {
 
 
 	public processRunOutput(rrr: RecombinedRuleResults): AnyJson {
-		const {minSev, summaryMap, results} = rrr;
-		// If the results are an empty string, it means no violations were found.
-		if (results === '') {
-			// Build an appropriate message...
-			const msg = messages.getMessage('output.noViolationsDetected', [[...summaryMap.keys()].join(', ')]);
-			// ...log it to the console...
-			this.ux.log(msg);
-			// ...and return it for use with the --json flag.
-			return msg;
-		}
```

Comment:
Removing this shortcut return is the main change here. The logic below works fine with 0 violations, it now returns `[]` when there are no violations, instead of an empty string. 

Most of the other changes in this PR are a result of removing these lines. The other changes involve updating the automated tests.